containers-tests/benchmarks/Graph.hs

```diff
@@ -46,12 +45,27 @@ data Graph = Graph
 instance NFData Graph where
   rnf (Graph label g edges adj) = rnf label `seq` rnf g `seq` rnf edges `seq` rnf adj
 
+makeG :: String -> G.Bounds -> [G.Edge] -> Graph
+makeG label bnds edges =
+  let g = G.buildG bnds edges
+  in Graph label g edges [(u, u, vs) | (u, vs) <- assocs g]
+
 -- A graph with vertices [1..n] and m random edges.
 buildRandG :: Int -> Int -> Graph
-buildRandG n m = Graph label g (G.edges g) [(u, u, vs') | (u, vs') <- assocs g]
+buildRandG n m = makeG label (1, n) edges
   where
-    label = "n=" ++ show n ++ ",m=" ++ show m
+    label = "rand,n=" ++ show n ++ ",m=" ++ show m
     xs = randomRs (1, n) (mkStdGen 1)
     (us, xs') = splitAt m xs
     vs = take m xs'
-    g = G.buildG (1, n) (zip us vs)
+    edges = zip us vs
+
+buildStarG :: Int -> Graph
+buildStarG n = makeG label (1, n) [(1, i) | i <- [2..n]]
+  where
+    label = "star,n=" ++ show n
```

Comment:
Brief explanations of these shapes would be good.

Reply:
Done.